src/IO/TinyPlyLoader/TinyPlyFileLoader.cpp

```diff
@@ -73,123 +71,106 @@ FileData* TinyPlyFileLoader::loadFile( const std::string& filename ) {
+                               const std::vector<std::string> propertyKeys ) {
+        std::shared_ptr<tinyply::PlyData> ret;
+        try
+        { ret = file.request_properties_from_element( elementKey, propertyKeys ); }
+        catch ( const std::exception& e )
+        {
+            ret = nullptr;
+            LOG( logERROR ) << "[TinyPLY] " << e.what();
+        }
+        return ret;
+    };
+
+    auto vertBuffer {initBuffer( "vertex", {"x", "y", "z"} )};
+
+    // if there is no vertex prop, or their count is 0, then quit.
+    if ( !vertBuffer || vertBuffer->count == 0 )
     {
         delete fileData;
         LOG( logINFO ) << "[TinyPLY] No vertice found";
         return nullptr;
     }
 
-    fileData->m_geometryData.clear();
-    fileData->m_geometryData.reserve( 1 );
+    auto startTime {std::clock()};
 
-    static int nameId = 0;
     // a unique name is required by the component messaging system
+    static int nameId {0};
     auto geometry = std::make_unique<GeometryData>( "PC_" + std::to_string( ++nameId ),
                                                     GeometryData::POINT_CLOUD );
     geometry->setFrame( Core::Transform::Identity() );
 
-    std::shared_ptr<tinyply::PlyData> normalBuffer( nullptr ), alphaBuffer( nullptr ),
-        colorBuffer( nullptr );
-    try
-    { normalBuffer = file.request_properties_from_element( "vertex", {"nx", "ny", "nz"} ); }
-    catch ( const std::exception& e )
-    {
-        normalBuffer = nullptr;
-        LOG( logERROR ) << "[TinyPLY] " << e.what();
-    }
-    try
-    { alphaBuffer = file.request_properties_from_element( "vertex", {"alpha"} ); }
-    catch ( const std::exception& e )
-    {
-        alphaBuffer = nullptr;
-        LOG( logERROR ) << "[TinyPLY] " << e.what();
-    }
-    try
-    { colorBuffer = file.request_properties_from_element( "vertex", {"red", "green", "blue"} ); }
-    catch ( const std::exception& e )
-    {
-        colorBuffer = nullptr;
-        LOG( logERROR ) << "[TinyPLY] " << e.what();
-    }
-
-    std::clock_t startTime;
-    startTime = std::clock();
+    std::shared_ptr<tinyply::PlyData> normalBuffer {initBuffer( "vertex", {"nx", "ny", "nz"} )};
+    std::shared_ptr<tinyply::PlyData> alphaBuffer {initBuffer( "vertex", {"alpha"} )};
+    std::shared_ptr<tinyply::PlyData> colorBuffer {
+        initBuffer( "vertex", {"red", "green", "blue"} )};
 
+    // read buffer data from file content
     file.read( ss );
 
-    {
-        const size_t numVerticesBytes = vertBuffer->buffer.size_bytes();
-        std::vector<Eigen::Matrix<float, 3, 1, Eigen::DontAlign>> verts( vertBuffer->count );
-        std::memcpy( verts.data(), vertBuffer->buffer.get(), numVerticesBytes );
-        geometry->setVertices( verts );
-    }
+    std::vector<Eigen::Matrix<float, 3, 1, Eigen::DontAlign>> verts( vertBuffer->count );
+    std::memcpy( verts.data(), vertBuffer->buffer.get(), vertBuffer->buffer.size_bytes() );
+    geometry->setVertices( verts );
```

Comment:
Out of scope for this PR : data are copied twice here : 
- First by the `std::memcpy` from `vertBuffer` to `verts`
- Second by `setVertices` from`verts` to `geometry->m_vertex`

This can result in performance issues when loading huge dataset. I understand that the second copy takes care of alignment and type promotion (i.e. from float to Scalar) but this second copy is expensive in term of memory allocation/reallocation (it uses a `back_inserter` into a vector that is not correctly sized most of the time.